File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxProfit = 0

        for i in range(1, len(prices)):
            if minPrice > prices[i]:
                minPrice = prices[i]
            currProfit = prices[i] - minPrice
            if currProfit > maxProfit:
                maxProfit = currProfit
        
        return maxProfit
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minPrice = prices[0];
    let maxProfit = 0;

    for (let i=1; i<prices.length; i+=1){
        if (minPrice > prices[i]){
            minPrice = prices[i];
        }
        const currProfit = prices[i] - minPrice;
        if (currProfit > maxProfit){
            maxProfit = currProfit;
        }
    }

    return maxProfit;
};